templates/dex-k8s-authenticator.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: kubeaddons.mesosphere.io/v1alpha1
+kind: Addon
+metadata:
+  name: dex-k8s-authenticator
+  namespace: kubeaddons
+spec:
+  kubernetes:
+    minSupportedVersion: v1.14.0
+  chartReference:
+    chart: raw.githubusercontent.com/dex-k8s-authenticator
+    # TODO(mh): If this is going to be used figure out the production URL
```

Comment:
If its agreeable for you @mhrabovcin, I would like to have a JIRA created for this TODO so that we can better keep track of this need (currently, we're not allowed to utilize github issues for kubeaddons work organization). Does that sound :heavy_check_mark: ?